### PR TITLE
DOC: linalg: various doc improvements

### PR DESCRIPTION
**Pull request by JozsefKutas**
#### What does this implement/fix?
Various minor clarifications/corrections to the docstrings for functions in `linalg`.

### CODE REVIEW

**Comment by JozsefKutas** [comment]
A quick summary of changes:
- `eigh` and `eigvalsh` may return no eigenvalues if the interval `subset_by_value` does not contain any eigenvalues. The documentation is updated to reflect this.
- Added docs for `eigh_tridiagonal` parameter `eigvals_only`.
- Fixed documented dimensions of `piv` array returned by `lu_factor`.
- Updated documentation of `qr` results to cover option `mode=raw`.
- Some minor formatting changes for consistency.

**Review by melissawm — APPROVED** [review]
These look like an improvement to me, thanks @JozsefKutas !

My only observation would be whether we want to add a link to https://www.netlib.org/lapack/lug/node69.html to make clearer how to recover Q and R from `mode='raw'`.

File: scipy/linalg/_decomp.py [hunk 1]
```diff
@@ -729,7 +730,7 @@ def eig_banded(a_band, lower=False, eigvals_only=False, overwrite_a_band=False,
         multiplicity.
     v : (M, M) float or complex ndarray
         The normalized eigenvector corresponding to the eigenvalue w[i] is
-        the column v[:,i].
+        the column v[:,i]. Only returned if ``eigvals_only=False``.
```
Comment (ilayn) [inline scipy/linalg/_decomp.py:733]: You don't have to warn for default behavior. `eigvals_only` is `True` by default.

File: scipy/linalg/_decomp.py [hunk 2]
```diff
@@ -1219,6 +1220,9 @@ def eigh_tridiagonal(d, e, eigvals_only=False, select='a', select_range=None,
         The diagonal elements of the array.
     e : ndarray, shape (ndim-1,)
         The off-diagonal elements of the array.
+    eigvals_only : bool, optional
+        Compute only the eigenvalues and no eigenvectors.
+        (Default: calculate also eigenvectors)
```
Comment (ilayn) [inline scipy/linalg/_decomp.py:1225]: Default is `False` not `calculate also eigenvectors`

File: scipy/linalg/_decomp.py [hunk 3]
```diff
@@ -1258,7 +1262,7 @@ def eigh_tridiagonal(d, e, eigvals_only=False, select='a', select_range=None,
         multiplicity.
     v : (M, M) ndarray
         The normalized eigenvector corresponding to the eigenvalue ``w[i]`` is
-        the column ``v[:,i]``.
+        the column ``v[:,i]``. Only returned if ``eigvals_only=False``.
```
Comment (ilayn) [inline scipy/linalg/_decomp.py:1265]: Same here no need for warning the default case

File: scipy/linalg/_decomp.py [hunk 4]
```diff
@@ -351,10 +351,11 @@ def eigh(a, b=None, lower=True, eigvals_only=False, overwrite_a=False,
     Returns
     -------
     w : (N,) ndarray
-        The N (1<=N<=M) selected eigenvalues, in ascending order, each
+        The N (N<=M) selected eigenvalues, in ascending order, each
         repeated according to its multiplicity.
     v : (M, N) ndarray
-        (if ``eigvals_only == False``)
+        The normalized eigenvector corresponding to the eigenvalue ``w[i]`` is
+        the column ``v[:,i]``. Only returned if ``eigvals_only=False``.
```
Comment (ilayn) [inline scipy/linalg/_decomp.py:358]: Same default warning that is not probably not needed.

File: scipy/linalg/_decomp.py [hunk 5]
```diff
@@ -976,7 +977,7 @@ def eigvalsh(a, b=None, lower=True, overwrite_a=False,
     Returns
     -------
     w : (N,) ndarray
-        The ``N`` (``1<=N<=M``) selected eigenvalues, in ascending order, each
+        The N (N<=M) selected eigenvalues, in ascending order, each
```
Comment (ilayn) [inline scipy/linalg/_decomp.py:980]: You need those double backticks to mark things as inline code

Comment (JozsefKutas) [reply scipy/linalg/_decomp.py:980]: I think I was just following `eigh` here, which doesn't use backticks. In general it seemed a bit random to me where double backticks are used and where they're not.

File: scipy/linalg/_decomp.py (hunk 2 above)
Comment (JozsefKutas) [reply scipy/linalg/_decomp.py:1225]: This was copied from `eig_banded`, but sure.

**Comment by JozsefKutas** [comment]
Thanks for the feedback @ilayn. I'm happy to tweak things, but since this has been merged I'm not sure it makes sense to reopen. I was mostly just copy & pasting docstring snippets from one function to another where bits seemed to be missing.

@melissawm - Sorry for not following up on this, I was busy with work and didn't get around to making the change.

**Comment by ilayn** [comment]
Somehow I completely missed that this was already merged. Sorry about that.

**Pull request merged**